nemo_aligner/metrics/common.py

```diff
@@ -0,0 +1,98 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A couple of common metrics as well as a handler class to provide a unified interface.
+
+from typing import Dict
+
+import hydra
+import torch
+from omegaconf import DictConfig
+from torchmetrics import Metric
+
+
+class InferenceMetricsHandler:
+    """A wrapper around metrics objects that will call update/compute/reset on all registered metrics.
+
+    If metrics_config is None, then all methods become no-ops and compute will return an empty dict.
+    """
+
+    def __init__(self, metrics_config: DictConfig):
+        if metrics_config is None:
+            metrics_config = {}
+        self.metrics = hydra.utils.instantiate(metrics_config)
+
+    def has_metrics(self) -> bool:
+        """Returns True if there are metrics to compute."""
+        return len(self.metrics) > 0
+
+    def update(self, batch: Dict, generation_output: Dict):
+        """Calling .update on all metrics.
+
+        Batch and generation output are coming directly from
+        validation dataloader and model.generate respectively.
+        """
+        for metric in self.metrics.values():
+            metric.update(batch, generation_output)
+
+    def compute(self) -> Dict[str, float]:
+        """Returns a dictionary with finalized metric values."""
+        return {name: metric.compute() for name, metric in self.metrics.items()}
+
+    def reset(self):
+        """Will reset state of all metrics to prepare for the next validation run."""
+        for metric in self.metrics.values():
+            metric.reset()
+
+
+def get_prediction(idx, batch, generation_output):
+    """A helper function to get predicted output for a typical dataloader format."""
+    elem_metadata = batch["metadata"][idx]
+    # need to remove original prompt from the generated output as it's always included
+    if "conversations" in elem_metadata:  # chat format
+        # last one is the target, so splitting on the one before that
+        pred_output = generation_output["sentences"][idx].split(elem_metadata["conversations"][-2]["value"])[-1]
+    else:  # input-output format
+        pred_output = generation_output["sentences"][idx][len(batch["metadata"][idx]["input"]) :]
+    return pred_output
+
+
+def get_target(idx, batch):
+    """A helper function to get expected target output for a typical dataloader format."""
+    elem_metadata = batch["metadata"][idx]
+    if "conversations" in elem_metadata:  # chat format
+        target_output = elem_metadata["conversations"][-1]["value"]  # always comparing the last one
+    else:  # input-output format
+        target_output = elem_metadata["output"]
+    return target_output
+
+
+class ExactStringMatchMetric(Metric):
+    """Computing exact string match between predicted and target outputs."""
+
+    def __init__(self, *args, **kwargs):
```

Comment:
`args` and `kwargs` are unused, is that intended?

Reply:
nope, that's a mistake, thanks for catching this!